Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -36,11 +37,11 @@ internal class HttpCoreDiagnosticSourceListener : IObserver<KeyValuePair<string,
         private readonly PropertyFetcher deprecatedResponseFetcher = new PropertyFetcher("Response");
         private readonly PropertyFetcher deprecatedRequestGuidFetcher = new PropertyFetcher("LoggingRequestId");
         private readonly PropertyFetcher deprecatedResponseGuidFetcher = new PropertyFetcher("LoggingRequestId");
-        
+
         #endregion
 
-        private readonly ConcurrentDictionary<string, IOperationHolder<DependencyTelemetry>> pendingTelemetry =
-            new ConcurrentDictionary<string, IOperationHolder<DependencyTelemetry>>();
+        private readonly ConditionalWeakTable<HttpRequestMessage, IOperationHolder<DependencyTelemetry>> pendingTelemetry = 
```

Comment:
> Oh, but it would mean that dependency will be tracked anyway when the HttpRequestMessage (the key) is disposed, which is a good side effect

This is not correct, because at the time when HttpRequestMessage will be disposed, there will be no matching activity for the operation and OperationHolder.Disponse will do nothing